main.tf

```diff
@@ -109,7 +109,8 @@ data "template_file" "bucket_policy" {
             "Resource": $${cloudtrail_resources},
             "Condition": {
                 "StringEquals": {
-                    "s3:x-amz-acl": "bucket-owner-full-control"
+                    "s3:x-amz-acl": "bucket-owner-full-control",
+                    "s3:x-amz-server-side-encryption": $${cloudtrail_sse_options_list}
```

Comment:
Can `s3:x-amz-server-side-encryption` be a list as you have in the default definitions for the variable? My read is the header is key = value. Perhaps it would be easier to see what the resulting bucket policy would be.